include/seqan/index/find2_index_approx.h

```diff
@@ -0,0 +1,585 @@
+{
+    static constexpr std::array<Search<5>, 3> VALUE
+    {{
+        { {{1, 2, 3, 4, 5}}, {{0, 0, 0, 0, 4}}, {{0, 3, 3, 4, 4}} },
+        { {{2, 3, 4, 5, 1}}, {{0, 0, 0, 0, 3}}, {{2, 2, 3, 3, 4}} },
+        { {{5, 4, 3, 2, 1}}, {{0, 0, 0, 3, 3}}, {{0, 0, 4, 4, 4}} }
+    }};
+};
+
+template <typename TVoidType>
+constexpr std::array<Search<5>, 3> SearchSchemes<3, 4, TVoidType>::VALUE;
+
+template <typename TVoidType>
+struct SearchSchemes<4, 4, TVoidType>
+{
+    static constexpr std::array<Search<5>, 3> VALUE
+    {{
+        { {{1, 2, 3, 4, 5}}, {{0, 0, 0, 0, 4}}, {{0, 3, 3, 4, 4}} },
+        { {{2, 3, 4, 5, 1}}, {{0, 0, 0, 0, 4}}, {{2, 2, 3, 3, 4}} },
+        { {{5, 4, 3, 2, 1}}, {{0, 0, 0, 3, 4}}, {{0, 0, 4, 4, 4}} }
+    }};
+};
+
+template <typename TVoidType>
+constexpr std::array<Search<5>, 3> SearchSchemes<4, 4, TVoidType>::VALUE;
+
+// Given the blocklengths (absolute, not cumulative values), assign it to all
+// Searches in a SearchScheme. The order of blocklength has to be from left to
+// right (regarding blocks)
+template <size_t nbrBlocks, size_t N>
+inline void _schemeSearchSetBlockLength(std::array<Search<nbrBlocks>, N> & ss,
+                                        std::vector<uint32_t> const & blocklength)
+{
+    for (Search<nbrBlocks> & s : ss)
+        for (uint8_t i = 0; i < s.blocklength.size(); ++i)
+            s.blocklength[i] = blocklength[s.pi[i]-1] + ((i > 0) ? s.blocklength[i-1] : 0);
+}
+
+// requires blocklength to be already set!
+template <size_t nbrBlocks, size_t N>
+inline void _schemeSearchInit(std::array<Search<nbrBlocks>, N> & ss)
+{
+    // check whether 2nd block is on the left or right and choose initialDirection accordingly
+    // (more efficient since we do not have to switch directions and thus have better caching performance)
+    // for that we need to slightly modify search()
+    for (Search<nbrBlocks> & s : ss)
+    {
+        s.startPos = 0;
+        for (uint8_t i = 0; i < s.pi.size(); ++i)
+            if (s.pi[i] < s.pi[0])
+                s.startPos += s.blocklength[i] - ((i > 0) ? s.blocklength[i-1] : 0);
+    }
+}
+
+template <size_t nbrBlocks, size_t N>
+inline void _schemeSearchComputeFixedBlocklength(std::array<Search<nbrBlocks>, N> & ss, uint32_t const needleLength)
+{
+    uint8_t blocks = ss[0].pi.size();
+    uint32_t blocklength = needleLength / blocks;
+    uint8_t rest = needleLength - blocks * blocklength;
+    std::vector<uint32_t> blocklengths;
+    for (uint8_t i = 0; i < blocks; ++i)
+        blocklengths.push_back(blocklength + (i < rest));
+
+    _schemeSearchSetBlockLength(ss, blocklengths);
+    _schemeSearchInit(ss);
+}
+
+template <typename TDelegate, typename TText, typename TIndex, typename TIndexSpec, typename TText2, size_t nbrBlocks,
+          typename TDir>
+inline void _schemeSearchDeletion(TDelegate & delegate,
+                                  Iter<Index<TText, BidirectionalIndex<TIndex> >, VSTree<TopDown<TIndexSpec> > > iter,
```

Comment:
this interface is still affected by the issue @rrahn reported, that if the args are on individual lines, the types should be, too